hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/stats/ColumnStatsIndices.java

```diff
@@ -300,12 +298,7 @@ private static List<RowData> readColumnStatsIndexByColumns(
     return records.collectAsList().stream().parallel().map(record -> {
           // schema and props are ignored for generating metadata record from the payload
           // instead, the underlying file system, or bloom filter, or columns stats metadata (part of payload) are directly used
-          GenericRecord genericRecord;
-          try {
-            genericRecord = (GenericRecord) record.getData().getInsertValue(null, null).orElse(null);
-          } catch (IOException e) {
-            throw new HoodieException("Exception while getting insert value from metadata payload");
-          }
+          GenericRecord genericRecord = (GenericRecord) record.getData().toHoodieMetadataRecord().orElse(null);
```

Comment:
Given that this still calls the `HoodieMetadataPayload` APIs which implements `HoodieRecordPayload`, shall we keep these as is?  Ideally, once we migrate the metadata payload to new merger API, these can be got rid of.

Reply:
Yes, we can leave it as is. Sound like it will not block rfc-46. Will undo the change for it.

Reply:
Just checked the files; if we leave it as is, then there is nothing to change for this PR.